lib/src/core/message/guild_emoji.dart

```diff
@@ -14,41 +15,121 @@ abstract class IBaseGuildEmoji implements SnowflakeEntity, IEmoji {
   /// Returns cdn url to emoji
   String get cdnUrl;
 
+  /// The name of the emoji.
+  String get name;
+
+  /// Whether this emoji is animated.
+  bool get animated;
+
   /// Creates partial emoji from given String or Snowflake.
-  factory IBaseGuildEmoji.fromId(Snowflake id) => GuildEmojiPartial(id);
+  factory IBaseGuildEmoji.fromId(Snowflake id) => GuildEmojiPartial({"id": id});
 }
 
 abstract class BaseGuildEmoji extends SnowflakeEntity implements IBaseGuildEmoji {
+  /// Reference to [NyxxWebsocket]
+  INyxx? get client;
+
   /// True if emoji is partial.
   @override
   bool get isPartial;
 
+  /// Whether this emoji is animated.
+  @override
+  bool get animated;
+
   /// Returns cdn url to emoji
   @override
-  String get cdnUrl => "https://cdn.discordapp.com/emojis/$id.png";
+  String get cdnUrl => "https://cdn.discordapp.com/emojis/$id.${animated ? 'gif' : 'png'}";
+
+  /// The name of the emoji.
+  @override
+  String get name;
 
   /// Creates an instance of [BaseGuildEmoji]
   BaseGuildEmoji(RawApiMap raw) : super(Snowflake(raw["id"]));
 
   @override
-  String formatForMessage() => "<:nyxx:$id>";
+  String formatForMessage() => "<${animated ? 'a' : ''}:$name:$id>";
 
   @override
-  String encodeForAPI() => id.toString();
+  String encodeForAPI() => '$name:$id';
 
   /// Returns encoded string ready to send via message.
   @override
   String toString() => formatForMessage();
 }
 
-abstract class IGuildEmojiPartial implements IBaseGuildEmoji {}
+abstract class IGuildEmojiPartial implements IBaseGuildEmoji {
+  /// Whether this emoji can be resolved to a [IGuildEmoji]
+  bool get isResolvable;
+}
+
+
+abstract class IResolvableGuildEmojiPartial implements IGuildEmojiPartial {
+  /// Resolves this [IResolvableGuildEmojiPartial] to [IGuildEmoji]
+  IGuildEmoji resolve();
+}
 
 class GuildEmojiPartial extends BaseGuildEmoji implements IGuildEmojiPartial {
+  /// Reference to [INyxxWebsocket]
+  @override
+  INyxx? client;
+
+  /// Whether this emoji can be resolved to a [IGuildEmoji]
+  @override
+  bool get isResolvable => this is IResolvableGuildEmojiPartial && client != null;
```

Comment:
that property is useless since end developer can check if objecys instance is resolvable.